File: solution/3/solution.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int j = 0;
        int longest = 0;
        int lenword = 0;
        int char_map[128] = {0};
        for(int i=0;i<s.length();i++){
            if(char_map[s[i]] == 0){
                char_map[s[i]]++;
                lenword ++;
                if(lenword>longest)longest = lenword;
            }else{
                while(s[i] != s[j]){
                    char_map[s[j]]--;
                    lenword--;
                    j++;
                }
                j++;
            }
        }
        return longest;
    }
};

File: solution/3/solution.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        j = 0
        result = 0
        maxlen = 0
        chardict = dict()
        for x in s:
            chardict[x] = chardict.get(x, 0) + 1
            if chardict[x] == 1:
                maxlen += 1
                if result < maxlen:result = maxlen
            else:
                while chardict[x]>1:
                    chardict[s[j]] -= 1
                    j += 1
                    maxlen -= 1
                maxlen += 1
        return result